src/domain/eventSearch/utils.tsx

```diff
@@ -263,30 +311,41 @@ export const getEventSearchVariables = ({
     keywordAnd.push('yso:p4354');
   }
 
-  const keywordsParamName = MAPPED_KEYWORD_TERMS[searchSource];
-  const mappedCategories: string[] = categories
-    .map((category) => MAPPED_CATEGORIES[searchSource][category])
-    .filter((e) => e);
-  const mappedHobbyTypes: string[] =
-    hobbyTypes
-      ?.map((hobbyType) => MAPPED_COURSE_HOBBY_TYPES[hobbyType])
-      .filter((e) => e) || [];
+  const categoriesParamName = MAPPED_KEYWORD_TERMS[searchSource];
+  let mappedCategories: string[] = [];
+  categories.map(
+    (category) =>
+      (mappedCategories = mappedCategories
+        .concat(MAPPED_CATEGORIES[searchSource][category]?.split(','))
+        .filter((e) => e))
+  );
+  let mappedHobbyTypes: string[] = [];
+  hobbyTypes?.map(
+    (hobbyType) =>
+      (mappedHobbyTypes = mappedHobbyTypes
+        .concat(MAPPED_COURSE_HOBBY_TYPES[hobbyType]?.split(','))
+        .filter((e) => e))
+  );
   const hasLocation = !isEmpty(divisions) || !isEmpty(places);
   const hasText = !isEmpty(text);
   // Combine and add keywords
 
+  //free text search
+  const freeTextSearchParam =
+    searchSource === EVENT_SEARCH_SOURCES.EVENTS
+      ? hasLocation
+        ? { localOngoingAnd: text }
+        : { allOngoingAnd: text }
+      : { allOngoingAnd: text };
+
   return {
-    ...(hasText &&
-      (hasLocation ? { localOngoingAnd: text } : { allOngoingAnd: text })),
+    ...(hasText && freeTextSearchParam),
     ...(hasLocation && { division: divisions.sort() }),
     end,
     include,
     isFree: isFree || undefined,
-    [keywordsParamName]: [
-      ...(keyword ? keyword : []),
-      ...mappedCategories,
-      ...mappedHobbyTypes,
-    ],
+    [categoriesParamName]: [...(keyword ? keyword : []), ...mappedCategories],
```

Comment:
can be simplified?

`[...(keyword ?? []), ...mappedCategories]`

Reply:
yes.. i am not used to ?? :) it was old logic, i just copied, but yea, can be refactored